core/application/bloodDonationWorkflow/BloodDonationService.ts

```diff
@@ -195,39 +195,37 @@ export class BloodDonationService {
   ): Promise<string> {
     try {
       const { seekerId, requestPostId, createdAt } = donorRoutingAttributes
-      const existingItem = await bloodDonationRepository.getItem(
+      const bloodDonationItem = await bloodDonationRepository.getItem(
         `${BLOOD_REQUEST_PK_PREFIX}#${seekerId}`,
         `${BLOOD_REQUEST_PK_PREFIX}#${createdAt}#${requestPostId}`
       )
-      if (existingItem === null) {
+      if (bloodDonationItem === null) {
         return 'Item not found.'
       }
-      if (
-        existingItem.status === DonationStatus.COMPLETED ||
-        existingItem.status === DonationStatus.EXPIRED
-      ) {
-        return "You can't update the donation request"
+      if (bloodDonationItem.status === DonationStatus.COMPLETED || bloodDonationItem.status === DonationStatus.EXPIRED) {
+        return 'You can\'t update the donation request'
       }
 
       const donorSearchItem = await donorSearchRepository.getItem(
         `${DONOR_SEARCH_PK_PREFIX}#${seekerId}`,
         `${DONOR_SEARCH_PK_PREFIX}#${createdAt}#${requestPostId}`
       )
       if (donorSearchItem === null) {
-        await donorSearchRepository.create(existingItem)
+        await donorSearchRepository.create(bloodDonationItem)
+      } else if (donorSearchItem.status === DonationStatus.COMPLETED) {
+        return 'Donor search is completed'
       }
 
       const retryCount = donorSearchItem?.retryCount ?? 0
-      const updateData: Partial<DonationDTO> = {
-        ...existingItem,
+      const updateData: Partial<DonorSearchDTO> = {
+        ...bloodDonationItem,
         id: requestPostId,
         retryCount: retryCount + 1
       }
 
       if (retryCount >= Number(process.env.MAX_RETRY_COUNT)) {
-        updateData.status = DonationStatus.EXPIRED
+        updateData.status = DonationStatus.COMPLETED
```

Comment:
should this really be COMPLETED

Reply:
this is not blood donation post entry. this the the status of donation search entry

Reply:
'The donor search process expired after the maximum retry limit is reached.' If the search process expired after trying, should it really be flagged as completed though?
It was not a successful request, COMPLETED tells me that it was successful

Reply:
Updated in ef9d4eba2a85f2333cfc52a99408d24bd8c51ede